### PR TITLE
Inline Calculators

### PR DESCRIPTION
**Pull request by benbarnett**
Introduces a generic javascript library to call for the calculator resulted via AJAX.

### CODE REVIEW

**Comment by andrewgarner** [comment]
Decent.

**Comment by andrewgarner** [comment]
Actually, scratch that. The behaviour if you submit an empty form is a but funky. Just recording a video to show!

**Comment by benbarnett** [comment]
Good point. Hm!

**Comment by andrewgarner** [comment]
It also suppresses the error states :see_no_evil:

**Comment by benbarnett** [comment]
@andrewgarner have rejigged the partials to include the form with the xhr return too, so now we get error messages too.

File: app/views/calculators/_form_input_submit.html.erb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <div class="form-group">
-  <input type="submit" class="button" value="Calculate" id="js-calculate">
+  <input type="submit" class="button calculator__submit js-calculate-submit" value="Calculate" id="js-calculate">
```
Comment (andrewgarner) [inline app/views/calculators/_form_input_submit.html.erb]: Do we still need the id of `js-calculate`?

File: app/controllers/calculators/take_whole_pot_controller.rb [hunk 2]
```diff
@@ -5,6 +5,10 @@ def show
       @income = params[:income]
 
       @calculator = TakeWholePotCalculatorForm.new(pot: @pot, income: @income)
+
+      render partial: 'calculators/take_whole_pot/calculator',
+             locals: { calculator: @calculator, pot: @pot, income: @income },
+             status: (@calculator.invalid? ? 400 : 200) if request.xhr?
```
Comment (andrewgarner) [inline app/controllers/calculators/take_whole_pot_controller.rb]: [Rails understands both numeric status codes and corresponding symbols](http://guides.rubyonrails.org/layouts_and_rendering.html).

I think using symbols improves the readability if you're not hot on your HTTP status codes.

```rb
status: (@calculator.valid? ? :ok : :bad_request)
```

File: app/views/calculators/take_whole_pot/_calculator.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<%= render 'form', calculator: @calculator, pot: @pot, income: @income %>
```
Comment (andrewgarner) [inline app/views/calculators/take_whole_pot/_calculator.html.erb]: You should use the locale variables passed to the partial not the instance variables defined by the controller. This allows the partial to be used in any context without it needing to know how it is being used.

```rb
<%= render 'form', calculator: calculator, pot: pot, income: income %>
<%= render 'result', calculator: calculator %>
```

**Comment by andrewgarner** [comment]
:+1:

**Pull request merged**